### ISSUE TITLE
existing appointment is flagged as available and keeps being "reset" to the same date/time stopping the scheduler

### ISSUE THREAD

**Issue by abondarev84** [phamleduy04/texas-dps-scheduler#148]
the automatic search finds as first available the same date that I have already set and as a result it cancels existing, sets the same one and exits.
updated to the latest version today.

**Comment by itsmateen** [comment]
change  **cancelIfExist** to **true** in config file

**Comment by abondarev84** [comment]
it is set to true.
what happens is that it first cancels an appointment and then sets the exact same one which causes it to exit (as intended I'm guessing).

but the new and old appointments are exactly the same one (time/date/place).

**Comment by phamleduy04** [comment]
sorry for the delay, do you still have logs?

**Comment by phamleduy04** [comment]
Because the way the bot is coded for now is:

- If you have an appointment -> the bot will check for a new one -> if the new one is available (doesn't check for the same day/same time for an old appointment) -> bot will cancel and book the new one.

What I think is that 2 appointments have the same date/time that are causing issues.

**Comment by abondarev84** [comment]
Since the appointment is in the past (I'm already done).  
I don't know if I can test again however, what was happening is that the exact same time and date (and location of course) was being set time and time again (I had like 5 or more emails about CANCELED and then RESERVED, or whatever they send) and I didn't realize at first that it was the exact same one so I was just restarting the program to get a closer date.

I actually saved the log (just now realized)
```
[07/30/2024 11:50:27] WARNING -> Invalid date in config.yml, using current date
[07/30/2024 11:50:27] Texas Scheduler v3.1.1 is starting...
[07/30/2024 11:50:27] Requesting Available Location....
[07/30/2024 11:50:27] WARNING -> You have an existing booking at Austin South 08/06/2024 09:00 AM
[07/30/2024 11:50:27] WARNING -> The bot will continue to run, but will cancel existing booking if it found a new one
[07/30/2024 11:50:29] Found cached location selection, using cached location selection
[07/30/2024 11:50:29] If you want to change location selection, please delete cache folder!
[07/30/2024 11:50:29] Checking Available Location Dates....
--------------------------------------------------------------------------------
[07/30/2024 11:50:40] Pflugerville Mega Center is not Available in around 0-60 days from 07/30/2024!
[07/30/2024 11:50:45] Georgetown is not Available in around 0-60 days from 07/30/2024!
[07/30/2024 11:50:48] Taylor is not Available in around 0-60 days from 07/30/2024!
--------------------------------------------------------------------------------
[07/30/2024 11:50:51] Austin North is not Available in around 0-60 days from 07/30/2024!
[07/30/2024 11:50:58] Austin South is Available on 8/6/2024 9:00 AM
[07/30/2024 11:50:59] Slot hold successfully
[07/30/2024 11:50:59] Booking slot....
[07/30/2024 11:50:59] Canceling existing booking 172636096
[07/30/2024 11:51:00] Canceled booking successfully
[07/30/2024 11:51:02] Slot booked successfully. Confirmation Number: 133590796
[07/30/2024 11:51:02] Visiting this link to print your booking:
[07/30/2024 11:51:02] https://public.txdpsscheduler.com/?b=133590796
```

**Comment by abondarev84** [comment]
my uneducated generic guess would be that there is somewhere a `<=` instead of `<` or something essentially similar to this.  
OR that the function that compares dates and "distance" to today implicitly gets a comparison result as true when "distance" to the new appointment is the same as current one.

**Comment by phamleduy04** [comment]
The bot doesn't do any comparison, will need to add this part to the code. I will fix it on the new release after #156. Thanks :)

**Comment by abondarev84** [comment]
for future users :-)

**Comment by abondarev84** [comment]
is this still relevant?
does it still function? (there was a note about txdps changing their api or something like that)

**Comment by phamleduy04** [comment]
I havnt work on this yet. Just make the bot working again
